### ISSUE TITLE
Failed to resolve entry for package "@zerodevx/svelte-toast"

### ISSUE THREAD

**Issue by rickexa** [zerodevx/svelte-toast#77]
I've just installed and imported this component in svelte but I get this error:

> Internal server error: Failed to resolve entry for package "@zerodevx/svelte-toast". The package may have incorrect main/module/exports specified in its package.json: No known conditions for "." specifier in "@zerodevx/svelte-toast" package

simply adding `import { SvelteToast } from '@zerodevx/svelte-toast'` to `+layout.svelte` page.

I tried to figure out what that could mean but I'm pretty new to svelte/sveltekit... so I'm writing here to seek for help.

Thanks.

**Comment by rickexa** [comment]
I indeed installed it as a dev dependency since in the readme is specified:
> npm i -D @zerodevx/svelte-toast

Anyway I've just tried the normal  `npm i` as you suggest and it seems to work properly.

Thank you very much.

**Comment by susnick** [comment]
Same for me. Uninstalling does not fix issue. I removed all modules and started from scratch. Issue persists.

**Comment by zerodevx** [comment]
Perhaps something changed in Svelte-world. @susnick, what's your setup? Are you using SvelteKit?

**Comment by susnick** [comment]
I am not using SvelteKit.  I use the vite template, svelte-ts for my project.

**Comment by zerodevx** [comment]
I just tried on a new `svelte-ts` instance and it seems to work?

https://stackblitz.com/edit/vitejs-vite-zqx6gs?file=src%2FApp.svelte&terminal=dev

**Comment by susnick** [comment]
Yes, I confirmed it works on fresh project. I tried installing on another project I had, and it gave same error, so there must be something else causing the issue. I just switched to using a different lib for toasts, and I have no issues.  If you want me to assist troubleshooting for this project, I will help, just point me in direction to go.

**Comment by zerodevx** [comment]
If you could provide a minimal repro on stackblitz or something, then it'll be much easier to debug.

**Comment by susnick** [comment]
So i put my project on stackblitz, and it worked fine. Ill have to spend some time to figure out why it wont work. Thanks for the assist.

**Comment by powellnorma** [comment]
> Same for me. Uninstalling does not fix issue.

Fwiw, I am facing the same problem, and am also using vite

**Comment by zerodevx** [comment]
@powellnorma could you do a minimal repro of the issue in stackblitz?

> Fwiw, I am facing the same problem, and am also using vite

Most people are using vite, but there are many flavours, like sveltekit, create-vite, or some other svelte-based starter template.

> What does this mean? What does this package specifiy that would need special handling?

Svelte-based template must use tooling that is svelte-aware, meaning when the bundler reads the exports from `package.json`:

```json
"exports": {
    ".": {
      "types": "./index.d.ts",
      "svelte": "./index.js"
    },
  //...
}
```

it understands that it's dealing with a svelte component.

**Comment by powellnorma** [comment]
> could you do a minimal repro

With the following I am able to reproduce it: https://github.com/powellnorma/toast-problem

Using `npm i` and `npm run dev`, I get:

```
✘ [ERROR] [plugin vite:dep-scan] Failed to resolve entry for package "@zerodevx/svelte-toast". The package may have incorrect main/module/exports specified in its package.json: No known conditions for "." entry in "@zerodevx/svelte-toast" package
```

Thank you for investigating!

**Comment by powellnorma** [comment]
Fyi, after specifying `"type": "module",` in `package.json` and updating `vite-plugin-svelte` + `vite` it worked

**Comment by Iuriy-Budnikov** [comment]
The same issue with vite

**Comment by zerodevx** [comment]
> The same issue with vite

Make sure your dependencies are updated, remove `/node_modules` and `package-lock.json`, then reinstall deps.

```
$ npx npm-check-updates -u
$ rm -rf node_modules
$ rm package-lock.json
$ npm i
```

**Comment by Jaydeep189** [comment]
> @rickexa did you accidentally install it as a dev dependency? I did that and got the same error, had to install with `yarn add` instead `yarn add -D` and then restart sveltekit

This Worked! Thanks

**Comment by kryptus36** [comment]
I ran into this as well - my solution was to:

pnpm i -D vite@latest  @sveltejs/vite-plugin-svelte@latest

**Comment by Iuriy-Budnikov** [comment]
Hey @zerodevx ,
It looks like an issue with using turbo repo. 
It can't resolve a path

"exports": {
    ".": {
      "types": "./index.d.ts",
      "svelte": "./index.js"
    },
  //...
}

**Comment by wetmarble** [comment]
I was able to resolve this issue by adding `"./package.json": "./package.json"` to the exports in package.json.

**Comment by meandrewandyou** [comment]
make sure you've updated `@sveltejs/vite-plugin-svelte` - that solved the issue in my case
